invisible_cities/cities/components.py

```diff
@@ -426,6 +474,32 @@ def ccwfs_to_zs(ccwf_sum, ccwf_sum_mau):
                 pkf.indices_and_wf_above_threshold(ccwf_sum    , thr_csum_s2).indices)
     return ccwfs_to_zs
 
+
+def compute_pe_resolution(rms, adc_to_pes):
+    return np.divide(rms                              ,
+                     adc_to_pes                       ,
+                     out   = np.zeros_like(adc_to_pes),
+                     where = adc_to_pes != 0          )
+
+
+def simulate_sipm_response(detector, run_number, wf_length, noise_cut, filter_padding):
+    datasipm      = load_db.DataSiPM (detector, run_number)
+    baselines     = load_db.SiPMNoise(detector, run_number)[-1]
+    noise_sampler = NoiseSampler(detector, run_number, wf_length, True)
+
+    adc_to_pes    = datasipm.adc_to_pes.values
+    thresholds    = noise_cut * adc_to_pes + baselines
+    single_pe_rms = datasipm.Sigma.values.astype(np.double)
+    pe_resolution = compute_pe_resolution(single_pe_rms, adc_to_pes)
+
+    def simulate_sipm_response(sipmrd):
+        wfs = sf.simulate_sipm_response(0, sipmrd[np.newaxis],
+                                        noise_sampler, adc_to_pes,
+                                        pe_resolution)
```

Comment:
I think it's maybe time to update sf.simulate_sipm_response so that we don't have to put dummy axes.

I'd change the function so instead of `event_indx, full_file_sipms` it just takes the sipm wfs for the event. That way you send `sipmrd` rather than `0, sipmrd[np.newaxis]`

Reply:
Done!